recycle-bin.c

```diff
@@ -19,22 +19,25 @@ int wmain(int argc, wchar_t **argv) {
 		return 1;
 	}
 
-	size_t len = argc + 1;
+	size_t len = argc;
```

Comment:
I realized the problem with this; `argc` is always at least 1. There is your final null byte. If `argc` is 2, indicating 1 provided path, then you have 1 for the path's null byte, and 1 for the final null byte (1+1=2).